broadinstitute/single_cell_portal_core
# adding caching to explore endpoint (SCP-3828)

This adds caching to the explore controller endpoint, with the hope of speeding up initial explore rendering.

This also shifts our study api cache invalidation strategy from 'attempt to be smart' to 'attempt to be careful'.  Our API endpoints are growing more complex and interdependent as we add new features to the explore tab, and tracking cache invalidation rules at a level more granular than the entire study requires quite deep knowledge.  So since study files themselves update relatively infrequently, and caching isn't essential for reasonable performance (especially since we have ClusterCacheService to regenerate automatically), I think opting for accuracy over speed is the right tradeoff.

TO TEST:
1. Open a study in the explore tab
2. reload the page
3. Observe in your console that the 'explore' and 'clusters' controller responses are cached
4. Update a cluster file (such as by changing the name, or changing the file)
5. after save and/or parse, reload the study
6. observe that the explore and clusters controller both are not cached, and instead hit the database
7. View a study with user annotations, not signed in
8. observe the user annotations are not shown in the annotation menu
9. sign in to the user who created the annotations
10. observe the user annotations are visible

(on a personal note, I think when I wrote `exclude?` on this test, it was the first time I ever got it right the first time on whether the method name is exclude, excludes, exclude?, or excludes?)
